python/contrib/dag/__init__.py

```diff
@@ -146,6 +146,22 @@ def delete_downstreams(self, node, graph=None):
                 self.delete_node_if_exists(edge)
         return deleted_nodes
 
+    # Added by the MOOSE group
+    def reverse_edges(self, graph=None):
```

Comment:
You might want a better name. Maybe `cloned_reversed`? From the name alone, it sounds like it might work on the current graph, not return a new graph.